libs/model/src/middleware/authorization.ts

```diff
@@ -182,10 +186,11 @@ async function buildAuth(
 /**
  * Checks if actor passes a set of requirements and grants access for all groups of the given topic
  */
-async function isTopicMember(
+async function hasTopicInteractionPermissions(
   actor: Actor,
   auth: AuthContext,
   action: GroupPermissionAction,
+  topicPermission: GroupTopicPermissionEnum,
```

Comment:
Seems like we can find a simpler generalization here, to avoid multiple permission enums and rules at the declaration layer... at the end of the day, these are all "user actions", and a single enum of actions should work.... we can encapsulate the internal complexity of model level validation (groups, topics, old model vs new model, etc)  in a validation map of actions to model validators... let's discuss